### PR TITLE
Splash Screen

### PR DESCRIPTION
**Pull request by sam33rdhakal**
Updates the design of splash screen. 
Only displays errors if the build is debug build.

### CODE REVIEW

File: splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt [hunk 1]
```diff
@@ -20,15 +24,38 @@ class SplashActivity : BaseViewModelActivity<SplashViewModel>() {
...
                 }
-                .bindToMain(percentText.text())
                 .disposedBy(disposeBag)
+
+        viewModel.percentage
+                .subscribe {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                        progressBar.setProgress((it * 100).toInt(), true)
+                    } else {
+                        progressBar.progress = it.toInt()
```
Comment (Cliabhach) [inline splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt]: ...not `(it * 100).toInt()`?

File: splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt [hunk 2]
```diff
@@ -20,15 +24,38 @@ class SplashActivity : BaseViewModelActivity<SplashViewModel>() {
...
+                    }
+                }.disposedBy(disposeBag)
+
+        setWindowFlag(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS, false)
```
Comment (Cliabhach) [inline splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt]: Move these two window-related operations to the top of the method, before the `super.onCreate()`. That way we avoid making an extra layout pass.

File: splash/src/main/res/drawable/splash_gradient.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+
+    <gradient
+        android:angle="90"
+        android:endColor="#2ea1a2"
```
Comment (Cliabhach) [inline splash/src/main/res/drawable/splash_gradient.xml]: Can these be switched to `@color/` resources?

File: splash/src/main/res/drawable/splash_gradient.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+        android:startColor="#1c515e" />
+
+    <corners android:radius="0dp" />
```
Comment (Cliabhach) [inline splash/src/main/res/drawable/splash_gradient.xml]: Isn't `0dp` the default radius? It should look the same if you delete this tag.

File: splash/src/main/res/layout/activity_splash.xml [hunk 5]
```diff
@@ -1,20 +1,94 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Cliabhach) [inline splash/src/main/res/layout/activity_splash.xml]: Please put `xmlns:android` declaration onto its own line.

File: splash/src/main/res/layout/activity_splash.xml [hunk 6]
```diff
@@ -1,20 +1,94 @@
...
+        android:id="@+id/supportedBy"
+        style="@style/MetadataSmallWhite"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/marginDouble"
+        android:layout_marginTop="@dimen/marginTriple"
+        android:layout_marginBottom="@dimen/marginStandard"
+        android:gravity="center"
+        android:maxLines="6"
+        android:text="Supported by"
```
Comment (Cliabhach) [inline splash/src/main/res/layout/activity_splash.xml]: Shouldn't this be moved into a string resource?

File: splash/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="splash">Splash</string>
```
Comment (Cliabhach) [inline splash/src/main/res/values/strings.xml]: If we're not using this for the activity's title, we should probably remove it.

File: splash/src/main/res/layout/activity_splash.xml [hunk 8]
```diff
@@ -1,20 +1,94 @@
...
+        android:textColor="@color/white"
+        app:layout_constraintBottom_toTopOf="@id/bloomBergLogo"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0"
+        app:layout_constraintStart_toStartOf="parent"
+        tools:ignore="HardcodedText" />
+
+
+    <ImageView
+        android:id="@+id/bloomBergLogo"
```
Comment (Cliabhach) [inline splash/src/main/res/layout/activity_splash.xml]: `bloomBerg` -> `bloomberg`.

File: splash/src/main/res/layout/activity_splash.xml [hunk 9]
```diff
@@ -1,20 +1,94 @@
...
-<LinearLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-    android:orientation="vertical"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    >
+    android:background="@drawable/splash_gradient"
```
Comment (Cliabhach) [inline splash/src/main/res/layout/activity_splash.xml]: Do we really need to specify this `android:background`? I thought the `android:windowBackground` in the manifest entry for `SplashActivity` would handle that.

File: splash/src/main/res/layout/activity_splash.xml (hunk 6 above)
Comment (sam33rdhakal) [reply splash/src/main/res/layout/activity_splash.xml]: Nah. Basically it is part of the logo which is always in English.

File: splash/src/main/res/layout/activity_splash.xml (hunk 8 above)
Comment (sam33rdhakal) [reply splash/src/main/res/layout/activity_splash.xml]: Updated.

File: splash/src/main/res/drawable/splash_gradient.xml (hunk 4 above)
Comment (sam33rdhakal) [reply splash/src/main/res/drawable/splash_gradient.xml]: Removed.

File: splash/src/main/res/drawable/splash_gradient.xml (hunk 3 above)
Comment (sam33rdhakal) [reply splash/src/main/res/drawable/splash_gradient.xml]: Extracted to colors.

File: splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt (hunk 2 above)
Comment (sam33rdhakal) [reply splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt]: Good idea. Done.

File: splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt (hunk 1 above)
Comment (sam33rdhakal) [reply splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt]: My bad. Totally forgot for else case. Updated.

File: splash/src/main/res/layout/activity_splash.xml (hunk 5 above)
Comment (sam33rdhakal) [reply splash/src/main/res/layout/activity_splash.xml]: Auto-indentation will reset the change and send it back to the first line with parent tag. Leaving it as it is. We could update our code style settings if you really want it.

Comment (Cliabhach) [reply splash/src/main/res/layout/activity_splash.xml]: I do want it, but it's not important now. We'll need to revisit indentation standards at some point anyway.

File: splash/src/main/res/layout/activity_splash.xml (hunk 6 above)
Comment (Cliabhach) [reply splash/src/main/res/layout/activity_splash.xml]: Odd. Guess it's fine then.

**Pull request merged**